base/task/sequence_manager/task_queue_impl.h

```diff
@@ -568,7 +570,11 @@ class BASE_EXPORT TaskQueueImpl {
   const scoped_refptr<GuardedTaskPoster> task_poster_;
 
   mutable base::internal::CheckedLock any_thread_lock_;
+  mutable base::internal::CheckedLock unordered_immediate_queue_lock_;
+  mutable base::internal::CheckedLock unordered_handler_lock_;
```

Comment:
Is having two separate unordered locks necessary?  If so can we have a comment describing why?  For simplicity it seems nicer if we could only have a single ordered lock and a single unordered lock, if possible.

Reply:
I thought about this hard, and I think you're right--the trick is to avoid wrapping any ordered lock access in the unordered lock (this is effectively what's leading to the deadlock.)  Updated, and with a warning comment.